controllers/users/register_v2.js

```diff
@@ -28,47 +28,87 @@ const registerV2 = async (req, res) => {
   const {data} = req.body;
   const {users, follows, local_community, topics} = data;
 
+  const checkUser = await User.findOne({
+    where: {
+      human_id: users.human_id
+    }
+  });
+
   let insertedObject = {};
+  let isUnverifiedUser = checkUser && checkUser.verified_status === 'UNVERIFIED';
 
-  /**
-   * Inserting data to Postgres DB
-   */
-  try {
-    insertedObject = await sequelize.transaction(async (t) => {
-      const user = await UsersFunction.register(User, users, t);
-      const anonymousUser = await UsersFunction.registerAnonymous(User, user?.user_id, t);
-
-      const topicRegistered = await TopicFunction.findAllByTopicIds(Topics, topics, t, true);
-      const locations = await LocationFunction.getAllLocationByIds(Locations, local_community, t);
-
-      return {
-        user,
-        anonymousUser,
-        topics: topicRegistered,
-        locations
-      };
-    });
-  } catch (e) {
-    console.log('error on sql transaction', e);
-    return res.status(500).json({
-      status: 'error on sql transaction',
-      code: 500,
-      message: e
-    });
+  if (isUnverifiedUser) {
+    let updatedUser;
+    try {
+      await checkUser.update(users);
+      updatedUser = await checkUser.save();
+    } catch (e) {
+      return res.status(500).json({
+        status: 'error on sql transaction',
+        code: 500,
+        message: e
+      });
+    }
+
+    const topicRegistered = await TopicFunction.findAllByTopicIds(Topics, topics);
+    const locations = await LocationFunction.getAllLocationByIds(Locations, local_community);
+
+    insertedObject = {
+      user: updatedUser,
+      anonymousUser: await UsersFunction.findAnonymousUserId(User, updatedUser.user_id, {
+        raw: false
+      }),
+      topics: topicRegistered,
+      locations
+    };
+  } else {
+    /**
+     * Inserting data to Postgres DB
+     */
+    try {
+      insertedObject = await sequelize.transaction(async (t) => {
+        const user = await UsersFunction.register(User, users, t);
+        const anonymousUser = await UsersFunction.registerAnonymous(User, user?.user_id, t);
+
+        const topicRegistered = await TopicFunction.findAllByTopicIds(Topics, topics, t, true);
+        const locations = await LocationFunction.getAllLocationByIds(Locations, local_community, t);
+
+        return {
+          user,
+          anonymousUser,
+          topics: topicRegistered,
+          locations
+        };
+      });
+    } catch (e) {
+      console.log('error on sql transaction', e);
+      return res.status(500).json({
+        status: 'error on sql transaction',
+        code: 500,
+        message: e
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, consider improving the error handling in this section as well.

```diff
-      console.log('error on sql transaction', e);
-      return res.status(500).json({
-        status: 'error on sql transaction',
-        code: 500,
-        message: e
-      });
+      console.error('Error on SQL transaction:', e);
+      return res.status(500).json({
+        status: 'error',
+        code: 500,
+        message: 'There was an error processing your request. Please try again later.'
+      });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->